src/interaction/ConstrainRG.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+  Copyright (C) 2017
+  Max Planck Institute for Polymer Research
+  
+  This file is part of ESPResSo++.
+  
+  ESPResSo++ is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  ESPResSo++ is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/
+
+// ESPP_CLASS
+#ifndef _INTERACTION_CONSTRAINRG_HPP
+#define _INTERACTION_CONSTRAINRG_HPP
+
+#include <cmath>
+#include <map>
+#include <boost/signals2.hpp>
+
+#include "mpi.hpp"
+#include "Potential.hpp"
+#include "FixedLocalTupleRgListInteractionTemplate.hpp"
+#include "esutil/Error.hpp"
+
+using namespace std;
+
+namespace espressopp {
+    namespace interaction {
+	/** This class provides methods to compute forces and energies of the
+	 *  ConstrainRG part.
+	 *  The code is based on G. Zhang's work. Reference in literature
+	 *  G. Zhang, L. A. Moreira, T. Stuehn, K. CH. Daoulas, K. Kremer,
+	 *  Macro Lett., 2014, 3, 198
+	 */
+	
+	class ConstrainRG : public PotentialTemplate< ConstrainRG > {
+	private:
+	    real k_rg;
+	    
+	public:
+	    static void registerPython();
+	    
+	    ConstrainRG(real _k_rg);
+	    
+	    ~ConstrainRG();
+	    
+	    
+/////////////////////////////////////////////////////////////////////////////////////////
+	    //  setters and getters
+	    void setK_rg(real _k_rg) {
+		k_rg = _k_rg;
+	    }
+	    real getK_rg() const { return k_rg; }
+/////////////////////////////////////////////////////////////////////////////////////////
+	    
+	    real _computeEnergy(real diff) {
+		
+		real energy = diff*diff;
+
+		return k_rg*energy;
+	    }
+	    
+	    Real3D _computeForce(Real3D diff,
+				 real rg_id,
+				 real diff_rg,
+				 long unsigned int N_Constrain) {
```

Comment:
I miss that p->mass() multiply the force term. Hence, I correct this. Actually, the force term described in this function is wrong. Then I also correct this formula.
$U_{rg} = k_{rg} \left(R_{g}^2 - {R_{g}^{ideal}}^2\right)^2$,
where $R_{g}^2 = \frac{1}{N} \sum^{N}_{i} (\vec{r}_i - \vec{R}_c)$ and $\vec{R}_c = \frac{1}{N} \sum^{N}_{i}$. Note that $\vec{R}_c$ is not the center of mass but center of a sub chain.
Then the force is represented as
$F_{rg} = -4 k_{rg} \left(R_{g}^2 - {R_{g}^{ideal}}^2\right) \times (\vec{r}_i - \vec{R}_c)$.